onecodex/api.py

```diff
@@ -103,6 +105,15 @@ def __init__(
         )
         self._client._fetch_schema(cache_schema=cache_schema)
         self._session = self._client.session
+
+        # Set backoff / retry strategy for 429s
```

Comment:
It'd be nice to have a unit test for the retry logic. `test_callback_retry_handling()` could be used as an example (it's testing upload retries)

Reply:
Actually if we remove the retry logic from `onecodex/lib/upload.py`, the existing test is probably good enough

Reply:
So, the test was using a mocked version of `session` so it was failing when I removed retry code in `onecodex/lib/upload.py` - I replaced that test with a test in `test_api_models` though